### DISCUSSION TITLE
Demo API Site seems unable to execute POSTs

### DISCUSSION THREAD

**Discussion by riversdev0** [firefly-iii/firefly-iii#6388] [Firefly III] [answered]
Hi -

I was exploring the [API documentation](https://api-docs.firefly-iii.org/) and noticed I can successfully execute `GET`s and `DELETE`s against the demo environment, but I can't get any `POST`s to work.  As an easy example, I tried to `store a new currency`.  So, I entered some details and executed the call.  It returns the code `undocumented`, with the following details: `TypeError: NetworkError when attempting to fetch resource.`

![image](https://user-images.githubusercontent.com/7363202/187239220-4c8f3831-a17d-4c78-ac7c-00cbfe0ee1ec.png)

I don't understand if this is something I'm doing wrong, some limitation of the demo environment, or some limitation with the API documentation.  Any idea what I can do to make this successful?

Thanks!

**Comment by JC5** [comment, marked as answer]
I see what you mean, it looks like a missing header. It should be fixed now :+1:

**Reply by riversdev0** [reply]
That's wonderful, thank you!  I am successfully able to add the currency using the API.  So I graduated to adding a transaction, and I get another `undocumented` return code, with some other details.  Is this related, or my own user error?

![image](https://user-images.githubusercontent.com/7363202/187272177-59c4dc00-3143-4cf1-a780-3930c16febbf.png)

**Reply by JC5** [reply]
Depending on your browser, there may be more info in the develop console.

**Reply by riversdev0** [reply]
This is interesting.  I made a formatting error with the JSON data.  When I removed the trailing commas, it submitted just fine.  But, when there is a formatting error, the browser inspector reveals the `422` coming back from the server, but the API Explorer just doesn't display that 422 correctly.

Thanks for the help!

**Reply by JC5** [reply]
That's weird! Glad to hear you found it 👍
